update-display.py

```diff
@@ -1,45 +1,222 @@
+import json
+import logging
+import math
 import os
+from os import environ
+import sys
 import textwrap
+import time
 import urllib.request
 import urllib.error
-import json
-import math
-import logging
 
-from PIL import Image, ImageFont, ImageDraw
+from font_amatic_sc import AmaticSC
+from font_caladea import Caladea
 from font_fredoka_one import FredokaOne
 from font_hanken_grotesk import HankenGrotesk
 from font_intuitive import Intuitive
-from font_source_serif_pro import SourceSerifPro
-from font_source_sans_pro import SourceSansPro
-from font_caladea import Caladea
 from font_roboto import Roboto
-from font_amatic_sc import AmaticSC
+from font_source_sans_pro import SourceSansPro
+from font_source_serif_pro import SourceSerifPro
+from PIL import Image, ImageFont, ImageDraw
+import arrow
+import geocoder
+
+icon_map = {
+    "clearsky": 1,
+    "cloudy": 4,
+    "fair": 2,
+    "fog": 15,
+    "heavyrain": 10,
+    "heavyrainandthunder": 11,
+    "heavyrainshowers": 41,
+    "heavyrainshowersandthunder": 25,
+    "heavysleet": 48,
+    "heavysleetandthunder": 32,
+    "heavysleetshowers": 43,
+    "heavysleetshowersandthunder": 27,
+    "heavysnow": 50,
+    "heavysnowandthunder": 34,
+    "heavysnowshowers": 45,
+    "heavysnowshowersandthunder": 29,
+    "lightrain": 46,
+    "lightrainandthunder": 30,
+    "lightrainshowers": 40,
+    "lightrainshowersandthunder": 24,
+    "lightsleet": 47,
+    "lightsleetandthunder": 31,
+    "lightsleetshowers": 42,
+    "lightsnow": 49,
+    "lightsnowandthunder": 33,
+    "lightsnowshowers": 44,
+    "lightssleetshowersandthunder": 26,
+    "lightssnowshowersandthunder": 28,
+    "partlycloudy": 3,
+    "rain": 9,
+    "rainandthunder": 22,
+    "rainshowers": 5,
+    "rainshowersandthunder": 6,
+    "sleet": 12,
+    "sleetandthunder": 23,
+    "sleetshowers": 7,
+    "sleetshowersandthunder": 20,
+    "snow": 13,
+    "snowandthunder": 14,
+    "snowshowers": 8,
+    "snowshowersandthunder": 21,
+}
+
+def create_mask(source):
```

Comment:
Right now the weather icons and masks are generated for Black / White displays. Then we could easily extend this to show other colors the screen supports.